packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -569,7 +569,7 @@ class MethodChannel {
     // TODO(jacksongardner): Use normal try/catch control flow here once
     // https://github.com/dart-lang/sdk/issues/55347 is fixed.
     final MethodCall call = codec.decodeMethodCall(message);
-    return handler(call)
+    return (() async => handler(call))()
```

Comment:
```suggestion
    return (() async => await handler(call))()
```

Do you want to add an await here? Not sure...

Reply:
Good question... maybe? I'll play around with it a bit. I am really hoping the dart2wasm bug can get fixed I can just revert this whole weird workaround before this PR merges, but we'll see.

Reply:
Actually I think they are equivalent, the futures get coalesced.

Reply:
Weird! Trying to imagine an `async` without an `await`. Hrm...

Reply:
Ömer has a fix out for the underlying issue, so we'll probably be able to remove this workaround altogether! https://dart-review.googlesource.com/c/sdk/+/361063

Reply:
This workaround was removed